.github/workflows/node.yml

```diff
@@ -132,19 +136,46 @@ jobs:
       - name: gas
         run: yarn workspace @hyperlane-xyz/core run gas-ci
 
-      - name: coverage
+      - name: base-coverage-cache
+        uses: actions/cache@v3
+        with:
+          path: './solidity/coverage/'
+          key: ${{ github.event.pull_request.base.sha }}
+
+      - name: persist-base-coverage
+        run: |
+          mv ./solidity/coverage/forge-lcov.info ./solidity/forge-lcov.base.info
+          mv ./solidity/coverage/lcov.info ./solidity/hardhat-lcov.base.info
+
+      - name: ref-coverage
         run: yarn workspace @hyperlane-xyz/core run coverage-ci
 
+      - name: ref-coverage-cache
+        uses: actions/cache@v3
+        with:
+          path: './solidity/coverage/'
+          key: ${{ github.sha }}
+
       - uses: osmind-development-org/lcov-reporter-action@v0.3.2
         with:
-          title: "Forge Coverage Report"
-          lcov-file: ./solidity/lcov.info
+          title: 'Forge Coverage Report'
+          lcov-file: ./solidity/coverage/forge-lcov.info
           lcov-base: ./solidity/forge-lcov.base.info
           delete-old-comments: true
 
       - uses: osmind-development-org/lcov-reporter-action@v0.3.2
         with:
-          title: "Hardhat Coverage Report"
+          title: 'Hardhat Coverage Report'
           lcov-file: ./solidity/coverage/lcov.info
           lcov-base: ./solidity/hardhat-lcov.base.info
           delete-old-comments: true
+
+      - name: check-coverage
+        run: |
+          sudo apt-get install lcov
+          cd ./solidity
+          export H_BASE=$(lcov --summary ./hardhat-lcov.base.info | grep "lines" | sed 's/.*lines......:\ //' | sed 's/% (.*//')
+          export H_REF=$(lcov --summary ./coverage/lcov.info | grep "lines" | sed 's/.*lines......:\ //' | sed 's/% (.*//')
+          export F_BASE=$(lcov --summary ./forge-lcov.base.info | grep "lines" | sed 's/lines......://' | sed 's/% (.*//')
+          export F_REF=$(lcov --summary ./coverage/forge-lcov.info | grep "lines" | sed 's/lines......://' | sed 's/% (.*//')
+          if [[ $H_REF < $H_BASE ]] && [[ $F_REF < $F_BASE ]]; then exit 1; fi
```

Comment:
surprised the action doesn't support failing on diff > x%

Reply:
It supports a minimum coverage, I suppose the best thing would be to plumb the base coverage into the minimum coverage, but I have it working this way so will keep it as is for now

Reply:
https://github.com/romeovs/lcov-reporter-action/issues/35

Reply:
@yorhodes what you linked above seems worse than caching? Going to merge now so I don't have to re-check in new artifacts but happy to follow up